app/templates/search/_categories.html

```diff
@@ -1,14 +1,47 @@
-<ul>
-  {% for category in categories %}
-    <li>
-      {% if category.link %}<a href="{{ category.link }}">{% endif %}{% if category.selected %}<strong>{% endif %}
-        {{ category.label }}
-      {% if category.selected %}</strong>{% endif %}{% if category.link %}</a>{% endif %}
-      {% if category.children and category.selected %}
-        {% with categories=category.children %}
-          {% include 'search/_categories.html' %}
-        {% endwith %}
+{%- macro top_level_link(text, url) -%}
+  <a href="{{ url }}" class="lot-filters__top-level-link"><span class="visually-hidden">Back to </span>{{ text }}</a>
+{%- endmacro -%}
+
+{%- macro list_item__unselected(text, url) -%}
```

Comment:
So you're saying just leave everything in the `category_list`?

I've pulled them out mostly so that the `selected` and `unselected` list items can handle their own presentation because the `category_list` is already a pretty chunky macro. 
My worry is that putting everything into the `category_list` will make it even harder to understand.